Sources/NIO/Embedded.swift

```diff
@@ -618,13 +618,24 @@ public final class EmbeddedChannel: Channel {
     /// - parameters:
     ///     - handler: The `ChannelHandler` to add to the `ChannelPipeline` before register or `nil` if none should be added.
     ///     - loop: The `EmbeddedEventLoop` to use.
-    public init(handler: ChannelHandler? = nil, loop: EmbeddedEventLoop = EmbeddedEventLoop()) {
+    public convenience init(handler: ChannelHandler? = nil, loop: EmbeddedEventLoop = EmbeddedEventLoop()) {
+        let handlers = handler.map { [$0] } ?? []
+        self.init(handlers: handlers, loop: loop)
+    }
+    
+    /// Create a new instance.
+    ///
+    /// During creation it will automatically also register itself on the `EmbeddedEventLoop`.
+    ///
+    /// - parameters:
+    ///     - handlers: The `ChannelHandler`s to add to the `ChannelPipeline` before register.
+    ///     - loop: The `EmbeddedEventLoop` to use.
+    public init(handlers: [ChannelHandler], loop: EmbeddedEventLoop = EmbeddedEventLoop()) {
         self.embeddedEventLoop = loop
         self._pipeline = ChannelPipeline(channel: self)
 
-        if let handler = handler {
-            // This will be propagated via fireErrorCaught
-            _ = try? _pipeline.addHandler(handler).wait()
+        if !handlers.isEmpty {
+            _ = try? _pipeline.addHandlers(handlers).wait()
         }
```

Comment:
```suggestion
        // `addHandlers` cannot fail here, the Channel is definitely not closed.
        try! self._pipeline.addHandlers(handlers).wait()
```

Reply:
- the `if` is superfluous
- shouldn't we also use the `.syncOperations` here? @glbrntt

Reply:
Its to avoid allocations

Reply:
@Davidde94 hence `.syncOperations`. But also: where do these allocs come from?

Reply:
@weissi the allocation is because `EmbeddedEventLoop` doesn't have a cached void future: `addHandlers`. The `if` returns us to the same state as before: only allocate if we actually add a handler. 

But yeah, sync operations is best here.